src/native/fs/fs_napi.cpp

```diff
@@ -38,20 +38,72 @@ static Napi::Value api(const Napi::CallbackInfo& info)
 struct FSWorker : public Napi::AsyncWorker
 {
     Napi::Promise::Deferred _deferred;
+    gid_t _orig_uid;
+    gid_t _req_uid;
+    uid_t _orig_gid;
+    uid_t _req_gid;
+    std::string _backend;
     int _errno;
     std::string _desc;
 
     FSWorker(const Napi::CallbackInfo& info)
         : AsyncWorker(info.Env())
         , _deferred(Napi::Promise::Deferred::New(info.Env()))
+        , _orig_uid(getuid())
+        , _orig_gid(getgid())
```

Comment:
why do we need every op to record this? Why not do that globally once in our fs_napi_init function?